components/fxa-client/src/http_client.rs

```diff
@@ -113,6 +135,41 @@ impl FxAClient for Client {
         self.make_oauth_token_request(config, body)
     }
 
+    fn refresh_token_with_session_token(
+        &self,
+        config: &Config,
+        session_token: &[u8],
+        scopes: &[&str],
+    ) -> Result<OAuthTokenResponse> {
+        let duplicate_url = config.auth_url_path("v1/session/duplicate")?;
+        let duplicate_key = derive_key_from_session_token(session_token)?;
+        let duplicate_body = json!({
+            "reason": "migration"
+        });
+        let duplicate_request =
+            HawkRequestBuilder::new(Method::Post, duplicate_url, &duplicate_key)
+                .body(duplicate_body)
+                .build()?;
```

Comment:
This "duplicate the sessionToken" thing is a one-time operation that should only be done during the initial migration, whereas I can imagine a future world in which we call this `refresh_token_with_session_token` method multiple times over the life of a program. What do you think about moving the duplicate logic into the migration-handling method to make it less likely that we duplicate our own sessionToken by accident?

Reply:
(For example, we might call it again in the future if we want to grant ourselves additional scopes)